macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
+	@objc public var contentTintColorPressed: NSColor?
 
-	/// Background color in rest state. Hover and pressed state colors will adjust accordingly using the systemEffect.
-	@objc public lazy var restBackgroundColor: NSColor  = fillColor {
-		didSet {
-			if restBackgroundColor != oldValue {
-				needsDisplay = true
+	/// Foreground text/image color when button is disabled.
+	@objc public var contentTintColorDisabled: NSColor?
+
+	/// Button foreground text/image color.  Setting this property determines the colors for the resting
+	/// (normal), pressed and disabled states by applying system effects.  The value returned when getting
+	/// this property will depend on which state currently applies to the button.  To set the state-specific
+	/// colors separately, use the `contentTintColorRest`, `contentTintColorPressed`,
+	/// and/or `contentTintColorDisabled` properties.
+	open override var contentTintColor: NSColor? {
+		get {
+			if !isEnabled {
+				return contentTintColorDisabled
+			} else if mouseDown {
+				return contentTintColorPressed
+			} else {
+				return contentTintColorRest
 			}
 		}
-	}
-	
-	private func disabledColor(for color: NSColor) -> NSColor {
-		return color.withSystemEffect(.disabled)
+		set {
+			contentTintColorRest = newValue
+			contentTintColorPressed = newValue == .clear ? newValue : newValue?.withSystemEffect(.pressed)
+			contentTintColorDisabled = newValue?.withSystemEffect(.disabled)
+			super.contentTintColor = newValue
+		}
 	}
 
-	private var trackingArea: NSTrackingArea?
+	/// Background fill color when button is not pressed or disabled.
+	@objc public var backgroundColorRest: NSColor?
 
-	private var mouseEntered = false
-	private var mouseDown = false
+	/// Background fill color when button is pressed.
+	@objc public var backgroundColorPressed: NSColor?
 
-	private var fillColor: NSColor {
-		return style == ButtonStyle.primaryFilled ? primaryColor : .clear
-	}
+	/// Background fill color when button is disabled.
+	@objc public var backgroundColorDisabled: NSColor?
 
-	private var outlineColor: NSColor {
-		let baseOutlineColor = style == ButtonStyle.primaryOutline ? primaryColor.withAlphaComponent(Button.outlineColorAlphaComponent) : .clear
-		if isEnabled {
-			return baseOutlineColor
-		} else {
-			return disabledColor(for: baseOutlineColor)
+	/// Button background fill color.  Setting this property determines the colors for the resting (normal),
+	/// pressed and disabled states by applying system effects.  The value returned when getting this
+	/// property will depend on which state currently applies to the button.  To set the state-specific
+	/// colors separately, use the `backgroundColorRest`, `backgroundColorPressed`, and/or
+	/// `backgroundColorDisabled` properties.
+	@objc public var backgroundColor: NSColor? {
+		get {
+			if !isEnabled {
+				return backgroundColorDisabled
+			} else if mouseDown {
+				return backgroundColorPressed
+			} else {
+				return backgroundColorRest
```

Comment:
So we don't need to have a different color for hover?

Reply:
Design says no.  For the sake of simplicity/YAGNI I removed the machinery for the hover state.